server/src/main/java/org/elasticsearch/repositories/RepositoriesModule.java

```diff
@@ -42,8 +44,10 @@ public RepositoriesModule(
         ClusterService clusterService,
         BigArrays bigArrays,
         NamedXContentRegistry namedXContentRegistry,
-        RecoverySettings recoverySettings
+        RecoverySettings recoverySettings,
+        TelemetryProvider telemetryProvider
     ) {
+        telemetryProvider.getMeter().registerLongCounter(METRIC_REQUESTS_COUNT, "repository request counter", "unit");
```

Comment:
Depending on how we name the metric, the registration need to happen in different places. After some back and forth, I think `repositories.requests.count` with dimension `{"repo_type": "s3", "repo_name": "xxx", "operation": "xxx", "purpose": "xxx"}` is better compared to something like `repositories.s3.requests.count`. Note the difference is where the repository type should go. I think it having it as dimension is better because it allows us to reuse the exactly same metric name (and likely associated chart) across different service providers (in future) and get the same data.

If we choose to have "s3" in the metric name, the registration will need to happen in at RepositoryPlugin level. If we further decide that "repo_name" should be part of the metric name, the registration then needs to happen at Repository level. In this case, I prefer to have less registration at high level.

Reply:
> I see the value in keeping the names same across repositories when possible, but I think it should be a best effort and we should register applicable metrics in each repository implementation

Yeah I agree. No need to coerce onto the same name if it does not apply.

Reply:
Than should it be moved to some S3 specific classes?

Reply:
Not really if we keep the same name, i.e. `repositories.requests.count`. Because you can only register a metric of the same name once. Having it register in s3 specific class means it cannot be registered in a Azure specific class which feels weird, i.e. why s3 should manage the registration while it is shared between different repositories. 

Assuming we keep the current metric name, my take is that we manage the registration here at the module level for metrics that could apply for mulitple types. Individual repository can pick and choose which one to use. Individual repo can also register their own _unique_ metric at the repository level, e.g. FsRepository.